internal/validators/validate.go

```diff
@@ -0,0 +1,80 @@
+package validators
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/stellar/go/strkey"
+)
+
+func NewValidator() *validator.Validate {
+	validate := validator.New()
+	_ = validate.RegisterValidation("public_key", publicKeyValidation)
+	validate.RegisterAlias("not_empty", "required")
+	return validate
+}
+
+func publicKeyValidation(fl validator.FieldLevel) bool {
+	addr := fl.Field().String()
+	return strkey.IsValidEd25519PublicKey(addr) || strkey.IsValidMuxedAccountEd25519PublicKey(addr)
+}
+
+func ParseValidationError(errors validator.ValidationErrors) map[string]interface{} {
+	fieldErrors := make(map[string]interface{})
+	for _, err := range errors {
+		fieldErrors[getFieldName(err)] = msgForFieldError(err)
+	}
+	return fieldErrors
+}
```

Comment:
There will be cases for when more than one validation fails, like in `validate:"required,public_key"`.

🤔 Do you think it's worth returning a slice of error messages, for each validation that fails in that field? Of would this library only return one error per field?

Reply:
It only returns one error per field, the order of the flags decides the validation order.

Reply:
It's fine to do incremental error handling in an API